Afternoon Projects/Unit 2 - Core Data & Delegates/Day 3/Task 1 - Part 1.md

```diff
@@ -81,3 +84,323 @@ Now you need to add a convenience initializer for your `Task` objects that match
+
+```
+override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+    if editingStyle == .delete {
+        
+        let task = TaskController.shared.tasks[indexPath.row]
+        
+        // Delete the row from the data source
+        TaskController.shared.delete(task)
+        
+        tableView.deleteRows(at: [indexPath], with: .fade)
+    }
+}
+    
+```
+##### Task Detail Table View Controller
+
+###### Storyboard
+
+Let's jump back into our storyboard and put the finishing touches on the detail view controller before jumping into code. Complete the following: 
+
+1. Click on the table view in the view hierarchy. 
+2. In the Attributes Inspector, increase the sections to three. Each section should only have one row. Give section one a title of 'Name.', section two 'Due', and section three 'Notes.' Because we are using a static table view controller, we can add UI elements straight to our table view cell. 
+3. Drag out a `UITextField` for both section one and two, and add a `UITextView` to section three. Constrain each element to be 4 points away from the leading, top, trailing, and bottom edges of the respective cell's content view.
+4. Drag a `UIDatePicker` into the view hierarchy under First Responder. We will use this date picker later to set the optional `due` property.
+5. Add a navigation item and a Save' `UIBarButtonItem` to the right side of the navigation bar.
+
+###### TaskDetailTableViewController.swift
+
+1. Drag outlets from your storyboard's view controller into your swift file. Make sure you include an outlet for the date picker.
+2. Add an `updateViews` function to your class that 1) unwraps your `Task` landing pad and 2) updates all of the UI elements on task detail view using the unwrapped landing pad's properties. Add a `didSet` to your `Task` landing pad and call both `loadViewIfNeeded` and `updateViews` in that order. `loadViewIfNeeded` loads the view if it hasn't already been loaded; if this function is not called, your app will crash when you get to the detail view.
+
+Next, let's tackle saving/updating. 
+
+1. Drag out an `IBAction` for your `saveBarButtonItem`.
+2. In `viewDidLoad`, assign the `dueTextField's` `inputView` to your date picker.
+3. In your `saveButtonTapped` function, safely unwrap the `nameTextField's` text and either a) create a new instance of `Task`, or b) update an existing task. Remember that `due` and `notes` are both optional. Also note that the data for `due` should come from your date picker, not the `dueTextField` itself. Finally, pop the view controller off the navigation stack. This should happen regardless of whether an update or creation is occuring.
```

Comment:
In your `saveButtonTapped` function, safely unwrap the `nameTextField's` text and using the value of `Task` handle creation and updating accordingly.